invenio_app_ils/stats/processors.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2025 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details
+
+"""ILS stats preprocessors."""
+
+
+def add_record_change_ids(doc):
+    """Add unique_id and aggregation_id to the doc."""
+
+    # We use this field to group by during aggregation
+    doc["aggregation_id"] = (
+        f"{doc.get('pid_type')}__{doc.get('method')}__{doc.get('user_id')}"
```

Comment:
```suggestion
        f"{doc['pid_type']}__{doc['method']}__{doc['user_id']}"
```
unless they are really optional, but in that case you might have `__None`. Same below.

Reply:
you are right. Changed it.